### PR TITLE
Tombstoned datastores 404 with tombstone info

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#2129](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2129)

Tombstoned datastores return a 404 when requested. The message lets users know that the 404 is because of a tombstoned datastore.

Future work:
- Persist the tombstone state in the summary so that tombstoned datastores are tombstoned when the summary loads

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:153]: Ha! Surprised there's no lint rule about this.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:336]: Is this because of that corner case where the Summarizer doesn't initialize the Trackers on connect?  Not sure it's worth highlighting so much in the "script" of the test.  I would ditch the `assert.doesNotReject` and just call `requestFluidObject`.  If it rejects it'll fail the test.

If you really do want to draw attention to the fact that this succeeds, leave it, but maybe give more explanation why, because it's pretty surprising (since we already waited `sweepTimeoutMs`).

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:346]: Old comment, right?

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:347]: Can we also test `handle.get`?  Seems a little weird I'll admit (since if you have the handle you probably stored it properly), but would be good to cover on account of any weird code people may write, or bugs we could have.

File: packages/runtime/container-runtime/src/dataStores.ts
Comment (markfields) [inline packages/runtime/container-runtime/src/dataStores.ts:439]: I think even after we implement full sweep, we may refer against the list of Deleted DataStores to give a distinct error here (v. above where it's just not found with no further context).  So I would want this message to not mention Tombstone, but say "Datastore removed by GC" or something like that.

@agarwal-navin this is what we were just talking about

File: packages/runtime/container-runtime/src/dataStores.ts
Comment (markfields) [inline packages/runtime/container-runtime/src/dataStores.ts:436]: Can you use `createResponseError` helper function? To avoid repeating things like the mimeType, stack getter, and consistent formatting of the error message.

**Review by markfields — COMMENTED** [review]
I love how concise this PR Is!  Left a few comments, will come back to approve after you have a chance to look.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:153]: Yep, there should have been

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:346]: Yep will fix

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:347]: Will do

File: packages/runtime/container-runtime/src/dataStores.ts
Comment (tyler-cai-microsoft) [reply packages/runtime/container-runtime/src/dataStores.ts:439]: Ok, I can update the message

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:336]: OK, I'll make it a comment, I thought this was important enough to illustrate in the test as it was a corner case

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:347]: See https://dev.azure.com/fluidframework/internal/_workitems/edit/405

We can log the "viaHandle" header of the request.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:336]: I just realized we already test this above so I'm going to remove this

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:300]: I simplified the test logic even more to add these handle tests and not drive myself crazy looking through all the different variations. 

Note: if someone makes a request and sets viaHandle = true, we will think it's a call made via handle. We thought about this problem and decided not to do anything about it, as a user can always figure out a way to make the request look like its internal

File: packages/runtime/container-runtime/src/dataStores.ts
Comment (markfields) [inline packages/runtime/container-runtime/src/dataStores.ts:423]: Can we have a follow-up to make this required?

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:250]: This error message is hard for me to read.  Can we try this instead?

```
`Context is tombstoned! Call site [${callSite}]`
```

There are some undocumented punctuation preferences I have too - no `:` (reserved for delineating an inner error message) and put `[]` around variable bits.  They're undocumented because I'm not sure they "work" as well as I hope, but trying them out when I can.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:400]:
```suggestion
    itExpects("Handle request for tombstoned datastores fails in summarizing container loaded before sweep timeout",
```

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts:392]: I was just in a meeting all about the difference between `resolveHandle` and `handle.get`.  I am honestly confused.  Is there a way to create a handle and call get on it?  Is this what RemoteObjectFluidHandle is or whatever? Let's discuss elsewhere.

**Review by markfields — APPROVED** [review]
LGTM!  Left some comments, mostly to follow-up on after merging

File: packages/runtime/container-runtime/src/dataStores.ts
Comment (tyler-cai-microsoft) [reply packages/runtime/container-runtime/src/dataStores.ts]: Made it required

**Pull request merged**
